src/Simulator/IncrementalRedstoneSimulator/DropSpenserHandler.h

```diff
@@ -18,6 +18,20 @@ class cDropSpenserHandler : public cRedstoneHandler
 	{
 	}
 
+	inline static bool IsActivated(NIBBLETYPE a_Meta)
+	{
+		return (a_Meta & E_META_DROPSPENSER_ACTIVATED) != 0;
+	}
+	inline static NIBBLETYPE SetActivationState(NIBBLETYPE a_Meta, bool IsOn)
+	{
+		a_Meta &= ~E_META_DROPSPENSER_ACTIVATED;  // clear the bit
+		if (IsOn)
+		{
+			a_Meta |= E_META_DROPSPENSER_ACTIVATED;
+		}
```

Comment:
Do we really want to twiddle the bit off and on? I think the clearing should be in an `else` branch of the `if` statement instead; and you could actually `return` the value directly.